src/core/server/saved_objects/migrations/state.ts

```diff
@@ -6,31 +6,15 @@
  * Side Public License, v 1.
  */
 
-import * as TaskEither from 'fp-ts/lib/TaskEither';
 import * as Option from 'fp-ts/lib/Option';
 import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 import { ControlState } from './state_action_machine';
 import { AliasAction } from './actions';
 import { IndexMapping } from '../mappings';
 import { SavedObjectsRawDoc } from '..';
 import { TransformErrorObjects } from '../migrations/core';
-import {
-  DocumentsTransformFailed,
-  DocumentsTransformSuccess,
-} from '../migrations/core/migrate_raw_docs';
 import { SavedObjectTypeExcludeFromUpgradeFilterHook } from '../types';
-
-export type MigrationLogLevel = 'error' | 'info' | 'warning';
-
-export interface MigrationLog {
-  level: MigrationLogLevel;
-  message: string;
```

Comment:
I thought it make sense to split the state-related types from the other types, especially as `src/core/server/saved_objects/migrations/types.ts` was already present, so I created this `state.ts` file.